packages/optimistic/src/collection.ts

```diff
@@ -200,11 +199,6 @@ export class Collection<T extends object = Record<string, unknown>> {
     this.optimisticOperations = new Derived({
       fn: ({ currDepVals: [transactions] }) => {
         const result = Array.from(transactions.values())
-          .filter(
-            (transaction) =>
-              transaction.state !== `completed` &&
```

Comment:
This PR exposed a potential bug where we'd lose the object -> key mapping if a transaction was removed but the synced data wasn't changed as then the object would revert to the synced version w/o updating the mapping.

So to prevent this from happening, we do need to loop over all transaction operations to get their keys (just applying the not completed/failed ones).